src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -209,12 +254,27 @@ public function setDefaults(array $defaults)
      * Returns true if {@link setDefault()} was called for this option.
      * An option is also considered set if it was set to null.
      *
-     * @param string $option The option name
+     * @param string|string[] $option The option name
      *
      * @return bool Whether a default value is set
      */
     public function hasDefault($option)
     {
+        if (\is_array($option) && 1 === count($option)) {
+            $option = $option[0];
+        }
+
+        if (\is_array($option)) {
```

Comment:
is it intended that this is not an "elseif" on the previous "if"? as is, this means we'll enter the block if `$option[0]` is an array (see lines above)